src/defaults.ts

```diff
@@ -1,43 +1,57 @@
 import { addDays, addMonths, addWeeks, endOfMonth, endOfWeek, startOfMonth, startOfWeek } from "date-fns";
 
+import { useDateRangeContext } from "./context";
 import { DefinedRange } from "./types";
 
-const getDefaultRanges = (date: Date): DefinedRange[] => [
-  {
-    label: "Today",
-    startDate: date,
-    endDate: date,
-  },
-  {
-    label: "Yesterday",
-    startDate: addDays(date, -1),
-    endDate: addDays(date, -1),
-  },
-  {
-    label: "This Week",
-    startDate: startOfWeek(date, { weekStartsOn: 1 }),
-    endDate: endOfWeek(date, { weekStartsOn: 1 }),
-  },
-  {
-    label: "Last Week",
-    startDate: startOfWeek(addWeeks(date, -1), { weekStartsOn: 1 }),
-    endDate: endOfWeek(addWeeks(date, -1), { weekStartsOn: 1 }),
-  },
-  {
-    label: "Last 7 Days",
-    startDate: addWeeks(date, -1),
-    endDate: addDays(date, -1),
-  },
-  {
-    label: "This Month",
-    startDate: startOfMonth(date),
-    endDate: endOfMonth(date),
-  },
-  {
-    label: "Last Month",
-    startDate: startOfMonth(addMonths(date, -1)),
-    endDate: endOfMonth(addMonths(date, -1)),
-  },
-];
+export const useDefinedRanges = (): DefinedRange[] => {
+  const date = new Date();
+  const { actions } = useDateRangeContext();
 
-export const defaultRanges = getDefaultRanges(new Date());
+  const actionList = [];
+
+  if (actions.today)
+    actionList.push({
+      label: actions.today,
+      startDate: date,
+      endDate: date,
+    });
+
+  if (actions.yesterday)
+    actionList.push({
+      label: actions.yesterday,
+      startDate: addDays(date, -1),
+      endDate: addDays(date, -1),
+    });
+  if (actions.thisWeek)
+    actionList.push({
+      label: actions.thisWeek,
+      startDate: startOfWeek(date, { weekStartsOn: 1 }),
+      endDate: endOfWeek(date, { weekStartsOn: 1 }),
+    });
+  if (actions.lastWeek)
+    actionList.push({
+      label: actions.lastWeek,
+      startDate: startOfWeek(addWeeks(date, -1), { weekStartsOn: 1 }),
+      endDate: endOfWeek(addWeeks(date, -1), { weekStartsOn: 1 }),
+    });
+  if (actions.last7Days)
+    actionList.push({
+      label: actions.last7Days,
+      startDate: addWeeks(date, -1),
+      endDate: addDays(date, -1),
+    });
+  if (actions.thisMonth)
+    actionList.push({
+      label: actions.thisMonth,
+      startDate: startOfMonth(date),
+      endDate: endOfMonth(date),
+    });
+  if (actions.lastMonth)
+    actionList.push({
+      label: actions.lastMonth,
+      startDate: startOfMonth(addMonths(date, -1)),
+      endDate: endOfMonth(addMonths(date, -1)),
+    });
```

Comment:
We don't want to do this calculation on every render. This is a good example where we should use `useMemo` hook.